dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/DispatchWorker.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.dolphinscheduler.server.master.runner;
+
+import org.apache.dolphinscheduler.dao.entity.TaskInstance;
+import org.apache.dolphinscheduler.plugin.task.api.enums.TaskExecutionStatus;
+import org.apache.dolphinscheduler.server.master.engine.task.client.ITaskExecutorClient;
+import org.apache.dolphinscheduler.server.master.engine.task.runnable.ITaskExecutionRunnable;
+import org.apache.dolphinscheduler.server.master.runner.queue.DelayEntry;
+import org.apache.dolphinscheduler.server.master.runner.queue.PriorityDelayQueue;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+public class DispatchWorker {
+
+    private final ITaskExecutorClient taskExecutorClient;
+
+    private final PriorityDelayQueue<DelayEntry<ITaskExecutionRunnable>> workerGroupQueue;
```

Comment:
Please add a `PriorityBasedTaskExecutionRunnableDelayEntry`, this entry should only compare priority rather than delay time.

Reply:
I have a question. If only priority is compared, then tasks that fail and need to be retried will not have any delay before being retried.

Reply:
![image](https://github.com/user-attachments/assets/4cd4f203-2701-4a22-abaf-5096796f9bd9)

Reply:
// todo: The current queue is flawed. When a high-priority task fails,
    // it will be delayed and will not return to the first or second position.
    // Tasks with the same priority will preempt its position.
    // If it needs to be placed at the front of the queue, the queue needs to be re-implemented.
    private final PriorityDelayQueue<PriorityAndDelayBasedTaskEntry> workerGroupQueue;